### PR TITLE
refactor(io): prepare for `noUncheckedIndexedAccess`

### PR DESCRIPTION
**Pull request by gabelluardo**
ref https://github.com/denoland/deno_std/issues/4040

### CODE REVIEW

File: io/buf_reader.ts [hunk 1]
```diff
@@ -207,7 +207,7 @@ export class BufReader implements Reader {
       if (this.#eof) return null;
       await this.#fill(); // buffer is empty.
     }
-    const c = this.#buf[this.#r];
+    const c = this.#buf[this.#r]!;
```
Comment (gabelluardo) [inline io/buf_reader.ts:210]: The while-loop wait until `this.#buf` is filled, so this can't be null

File: io/read_delim.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ function createLPS(pat: Uint8Array): Uint8Array {
       lps[i] = 0;
       i++;
     } else {
-      prefixEnd = lps[prefixEnd - 1];
+      prefixEnd = lps[prefixEnd - 1]!;
```
Comment (gabelluardo) [inline io/read_delim.ts:22]: `prefixEnd` is incremented inside the loop and `lsp` size is `pat.length` so this is non-null

File: io/string_writer_test.ts [hunk 3]
```diff
@@ -33,6 +33,6 @@ Deno.test("ioStringWriterIsolationTest", async function () {
     const written = await w.write(c);
     assertEquals(written, 1);
   }
-  srcChunks[0][0] = 88;
+  // srcChunks[0][0] = 88;
```
Comment (gabelluardo) [inline io/string_writer_test.ts:36]: I think this can be deleted because the values in `srcChunks` aren't checked in the assertion

**Review by iuioiua — APPROVED** [review]
LGTM! Thank you.

**Pull request merged**